### PR TITLE
Add cloudbuild.yaml

### PR DESCRIPTION
**Pull request by natarajaya**

### CODE REVIEW

**Comment by mrtyler** [comment]
LGTM

I'm not sure exactly what the context is but this looks like a worthwhile experiment.

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+steps:
+- name: 'hashicorp/terraform:light'
```
Comment (stepanstipl) [inline cloudbuild.yaml]: Actually one issue - can we do this with our Exekube image? That way the TF version running the fmt check will be consistent with the execution environment, especially as we don't have a version tag or anything pinning the version used (what if they release 0.12 new HCL?)

**Comment by natarajaya** [comment]
@amatas @mrtyler @stepanstipl Thanks for your review guys, but I honestly have no understanding how this PR got opened.

I was experimenting on my fork to enable Google Cloud Build integration as PR check:
https://github.com/natarajaya/gpii-infra/pull/5

With that we can have our `terraform-fmt-check` as PR check (to prevent you from merging PR with broken HCL formatting) instead of additional CI step (to punish you for merging PR with broken formatting). We can probably move our less expensive tests (unit tests for gcp and aws?) there as well!

Looks like everybody liked this idea, so I am going to close this PR and reopen a new one with full implementation.

File: cloudbuild.yaml (hunk 1 above)
Comment (natarajaya) [reply cloudbuild.yaml]: Yes, we can. One downside of that is that our exekube image is pretty heavy, which increases build times and also causes some extra charges (normally cloud build is free for open repos). We may want to push exekube image to Google Container Registry to improve this.

**Pull request closed without merging**